### ISSUE TITLE
Some PNG images are not correctly rendered when added to a PDF

### ISSUE THREAD

**Issue by cremor** [UglyToad/PdfPig#346]
I use the following (simplified) code to add images to a PDF file:

```cs
private static void AddImage(PdfDocumentBuilder outputDocument, string imageFile)
{
    double width;
    double height;

    using (var image = Image.FromFile(imageFile))
    {
        width = image.Width / image.HorizontalResolution * 72;
        height = image.Height / image.VerticalResolution * 72;
    }

    var page = outputDocument.AddPage(width, height);

    using (var imageStream = File.OpenRead(imageFile))
    {
        page.AddPng(imageStream, page.PageSize);
    }
}
```

This works fine for most images, but some PNG images are not correctly shown in the created PDF file. I have not yet figured out which PNGs work and which don't, but here are two sample files that don't work:

* https://user-images.githubusercontent.com/116314/124563968-d784a600-de40-11eb-8de0-39480732ad28.png
  This file was converted from a GIF by using `System.Drawing.Image.FromFile()` and `image.Save(path, ImageFormat.Png)`.
* https://www.a-trust.at/images/AtrustLogoSmall.png
  Note: This image also shows a second problem because it contains transparency. I've opened a separate issue for that: #345

I'm currently using version 0.1.5-alpha002 of PdfPig.

Here is a screenshot of how those two images are then shown in the PDF file:
![grafik](https://user-images.githubusercontent.com/116314/124564796-b53f5800-de41-11eb-9e4f-439c0a38142f.png)

**Comment by cremor** [comment]
FYI: Even after the fix for #345 the two test PNGs provided here still look the same in the PDF.

**Comment by EliotJones** [comment]
I didn't test with your images specifically but I ported a couple of fixes from the PNG library used internally https://github.com/EliotJones/BigGustave please let me know if either of them gets fixed by this.

**Comment by cremor** [comment]
I can confirm that the current master (at 889063f83ca3baeb894ca6be4e936ff3bf3ebc90) works fine with those two sample images. Thanks!

**Comment by ThomasSchattat** [comment]
I see the same issue when I extract PNG files from a PDF file (not insert into a PDF), the background is black instead of white. Has that also been resolved? I used the latest version 0.1.10 and my extracted images are unusable like that.

**Comment by BobLd** [comment]
@ThomasSchattat your issue is not related to this one. I believe this is related to transparency masks which are not supported yet. This is going the next addition to PdfPig
